skema/skema-rs/mathml/src/expression.rs

```diff
@@ -14,42 +23,1094 @@ enum Atom {
+        Mo(Operator::Multiply),
+        Mi("g".to_string()),
+        Mo(Operator::Subtract),
+        Mi("h".to_string()),
+    ]);
+    let mut pre_exp = PreExp {
+        op: Vec::<Operator>::new(),
+        args: Vec::<Expr>::new(),
+        name: "".to_string(),
+    };
+    pre_exp.op.push(Operator::Other("root".to_string()));
+    math_expression.to_expr(&mut pre_exp);
+    pre_exp.group_expr();
+    pre_exp.get_names();
+    let g = pre_exp.to_graph();
+    println!("{}", Dot::new(&g));
+}
+
+#[test]
+fn test_to_expr13() {
+    let math_expression = Mrow(vec![
+        Mi("a".to_string()),
+        Mo(Operator::Add),
+        Mi("b".to_string()),
+        Mo(Operator::Multiply),
+        Mi("c".to_string()),
+        Mo(Operator::Multiply),
+        Mi("a".to_string()),
+        Mo(Operator::Divide),
+        Mi("d".to_string()),
+        Mo(Operator::Subtract),
+        Mi("c".to_string()),
+        Mo(Operator::Multiply),
+        Mi("a".to_string()),
+        Mo(Operator::Subtract),
+        Mi("b".to_string()),
+    ]);
+    let mut pre_exp = PreExp {
+        op: Vec::<Operator>::new(),
+        args: Vec::<Expr>::new(),
+        name: "".to_string(),
+    };
+    pre_exp.op.push(Operator::Other("root".to_string()));
+    math_expression.to_expr(&mut pre_exp);
+    pre_exp.group_expr();
+    pre_exp.get_names();
+    let g = pre_exp.to_graph();
+    println!("{}", Dot::new(&g));
+}
+
+#[test]
+fn test_to_expr14() {
+    let math_expression = Mrow(vec![
+        Mi("a".to_string()),
+        Mo(Operator::Add),
+        Mi("b".to_string()),
+        Mo(Operator::Multiply),
+        Mi("c".to_string()),
+        Mo(Operator::Multiply),
+        Mrow(vec![
+            Mi("a".to_string()),
+            Mo(Operator::Subtract),
+            Mi("b".to_string()),
+        ]),
+    ]);
+    let mut pre_exp = PreExp {
+        op: Vec::<Operator>::new(),
+        args: Vec::<Expr>::new(),
+        name: "".to_string(),
+    };
+    pre_exp.op.push(Operator::Other("root".to_string()));
+    math_expression.to_expr(&mut pre_exp);
+    pre_exp.group_expr();
+    pre_exp.get_names();
+    let g = pre_exp.to_graph();
+    println!("{}", Dot::new(&g));
+}
+
+#[test]
+fn test_to_expr15() {
+    let math_expression = Mrow(vec![
+        Mi("a".to_string()),
+        Mo(Operator::Add),
+        Mi("b".to_string()),
+        Mo(Operator::Multiply),
+        Mi("c".to_string()),
+        Mo(Operator::Subtract),
+        Msqrt(Box::from(Mrow(vec![
+            Mi("a".to_string()),
+            Mo(Operator::Add),
+            Mi("b".to_string()),
+        ]))),
+    ]);
+    let mut pre_exp = PreExp {
+        op: Vec::<Operator>::new(),
+        args: Vec::<Expr>::new(),
+        name: "".to_string(),
+    };
+    pre_exp.op.push(Operator::Other("root".to_string()));
+    math_expression.to_expr(&mut pre_exp);
+    pre_exp.group_expr();
+    pre_exp.get_names();
+    let g = pre_exp.to_graph();
+    println!("{}", Dot::new(&g));
```

Comment:
Can you remove the `println!` calls in the test functions? They are fine for debugging purposes, but once you know the test passes, we don't need them.

Reply:
Got it! I will remove them.

Reply:
Thanks @ualiangzhang  - I see that you've removed the printing of the DOT output, but there are still some statements like `println!("Success!")` in the tests - could you remove them as well? Running `cargo test` will show us whether tests are successful, so we don't need to additionally print another success indicator.

Reply:
Got it! Will remove them.